File: src/main/java/io/spring/treative/tourist/Tourist.java
package io.spring.treative.tourist;

import io.spring.treative.flight.Flight;
import lombok.Data;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.sql.Timestamp;
import java.util.ArrayList;
import java.util.List;

@Data
@Entity
public class Tourist {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @NotNull
    private String firstName;
    @NotNull
    private String lastName;
    @Enumerated(value = EnumType.STRING)
    private Gender gender;
    @NotNull
    private String country;
    private String remarks;
    @NotNull
    private Timestamp dayOfBirth;
    @ManyToMany(mappedBy = "tourists")
    private List<Flight> flights = new ArrayList<>();
}

File: src/main/ui/src/app/model/tourist.ts
import {Gender} from "./gender";
import {Flight} from "./flight";
import DateTimeFormat = Intl.DateTimeFormat;

export class Tourist {

  id: number;
  firstName: string;
  lastName: string;
  gender: Gender;
  country: string;
  remarks: string;
  dayOfBirth: DateTimeFormat;
  flights: Flight[];
}
